squash_bot/list_timetable/commands.py

```diff
@@ -0,0 +1,179 @@
+import typing
+from urllib.parse import urljoin
+from datetime import datetime
+from dataclasses import dataclass
+
+import enum
+import requests
+
+from squash_bot.core import command as _command
+from squash_bot.core import command_registry, lambda_function
+from squash_bot.core.data import dataclasses as core_dataclasses
+from squash_bot.settings import base as settings_base
+
+
+TIMETABLE_API_DATETIME_FORMAT = "YYYY-MM-DDTHH:mm:ss+00:00"
+TIMETABLE_HEADER_DATETIME_FORMAT = "%d-%m"
+TIMETABLE_RESULT_DATETIME_FORMAT = TIMETABLE_HEADER_DATETIME_FORMAT + " HH:mm"
+
+RESULTS = "Results"
+AVAILABLE_SLOTS = "AvailableSlots"
+START = "start"
+RESOURCE_SCHEDULE_ID = "ResourceScheduleId"
+
+
+class ListTimetableOptionType(enum.Enum):
+    FROM_DATE = "from-date"
+    DAYS = "days"
+    TIME_OF_DAY = "time-of-day"
+
+
+class TimeOfDayType(enum.Enum):
+    MORNING = "Morning"
+    AFTERNOON = "Afternoon"
+    EVENING = "Evening"
+    ANY = "Any"
+
+
+@dataclass
+class TimetableResult:
+    start_datetime: datetime
+    available_slots: int
+    schedule_id: int
+
+    def __str__(self):
+        return f"{datetime.strftime(DISCORD_RESPONSE_DATETIME_FORMAT)}: {available_slots} slots available"
+
+
+@command_registry.registry.register
+class ListTimetableCommand(_command.Command):
+    name = "list-timetable"
+    description = "List the timetable between two datetimes"
+    options = (
+        _command.CommandOption(
+            name=ListTimetableOptionType.FROM_DATE.value,
+            description="Datetime string in the format DD-MM-YY",
+            type=_command.CommandOptionType.STRING,
+            required=True,
```

Comment:
As mentioned in discord, I think we could just have the user supply a number of days in the future and the code just assumes that the `from_date` is just today

I think we'll never want to find slots in the past and the API we're using only provides slots for the next 7 days anyway so the flexibility of allowing the user to provide a date will never be used imo

Reply:
Sure can do that

Reply:
Done